### PR TITLE
ci: test publishing before merging PR to help prevent deploy failure

### PR DESCRIPTION
**Pull request by levibostian**
This is an attempt at preventing [deploy failures](https://github.com/customerio/customerio-flutter/actions/runs/5187051663/jobs/9348935326) by running similar commands deployment does before we merge PRs. 

This PR also added a setup-flutter action to make code more re-usable across workflow files.

### CODE REVIEW

**Comment by levibostian** [comment]
Success! Looks like I was able to [reproduce](https://github.com/customerio/customerio-flutter/actions/runs/5415121571/jobs/9843112999?pr=60#step:6:391) the failure that happened in the past deployment. 

Next step now is to fix these 2 error messages (issues) that are thrown, get the CI to pass, then we can merge this PR. 

Anyone can help get these 2 error messages resolved.

File: .github/actions/setup-flutter/action.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      cache: true
+
+  - name: Install Flutter dependencies
+    run: flutter pub get
+    shell: bash 
+
+  # flutter analyze throws error with this file missing. 
+  # In future CI work for compiling sample apps, we can move this logic into another github workflow file with populating real credentials.
+  - name: Create credentials.env file for sample app 
+    run: cp example/credentials.env.example example/credentials.env
```
Comment (Shahroz16) [inline .github/actions/setup-flutter/action.yml:20]: We don't use this file anymore in `example`, the fact `credentials.env` still lives in repo is probably why its causing this issue.

**Review by Shahroz16 — APPROVED** [review]

**Comment by Shahroz16** [comment]
This might not be needed anymore, I pulled changes from this PR and left out something that I felt was unwanted, for example adding a version as part of the changelog. I didn't understand that fully, as this is already being done. 

Feel free to either close this PR or pull the changes from main, and compare if you still want to add the left out changes.

**Comment by levibostian** [comment]
@Shahroz16 all of [the code that is making changes such as changelog](https://github.com/customerio/customerio-flutter/pull/60/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R25-R34) is a hack for flutter dry-run deployments. 

[See my new comment](https://github.com/customerio/customerio-flutter/pull/66#discussion_r1273511726) I left in the new PR to understand.

**Comment by levibostian** [comment]
Closing due to age.

**Pull request closed without merging**